app/pb/search.proto

```diff
@@ -0,0 +1,135 @@
+syntax = "proto3";
+
+package org.couchers.api.search;
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/wrappers.proto";
+
+import "pb/api.proto";
+import "pb/communities.proto";
+import "pb/discussions.proto";
+import "pb/groups.proto";
+import "pb/pages.proto";
+
+service Search {
+  // This is a secure service: a user needs to be authenticated and not jailed to call functions here, refer to auth.proto and jail.proto
+
+  rpc Search(SearchReq) returns (SearchRes) {
+    // Search all content
+  }
+
+  rpc UserSearch(UserSearchReq) returns (UserSearchRes) {
+    // Search for users in particular with extra filters, e.g. hosts
+  }
+}
+
+message Area {
+  double lat = 1;
+  double lng = 2;
+  double radius = 3; // m
+}
+
+message SearchReq {
+  string query = 1;
```

Comment:
Please document the format of the search string:

* Are AND and OR parsed and interpreted?
* Will minus in front of a word remove those results from the hits?
* Will quotation marks force a literal search for that string?

Reply:
Added a comment